diff -Naur ../guacamole-azure-windowsvm/terraform/data.tf terraform/data.tf
--- ../guacamole-azure-windowsvm/terraform/data.tf	2022-09-13 10:40:05.000000000 +0100
+++ terraform/data.tf	2022-10-03 16:30:11.000000000 +0100
@@ -17,6 +17,12 @@
   resource_group_name  = data.azurerm_resource_group.ws.name
 }
 
+data "azurerm_subnet" "webapps" {
+  name                 = "WebAppsSubnet"
+  virtual_network_name = data.azurerm_virtual_network.ws.name
+  resource_group_name  = data.azurerm_resource_group.ws.name
+}
+
 data "azurerm_key_vault" "ws" {
   name                = local.keyvault_name
   resource_group_name = data.azurerm_resource_group.ws.name
@@ -26,3 +32,8 @@
   name                = "guacamole-${var.tre_id}-ws-${local.short_workspace_id}-svc-${local.short_parent_id}"
   resource_group_name = data.azurerm_resource_group.ws.name
 }
+
+data "azurerm_private_endpoint_connection" "airlock_export_inprogress_pe" {
+  name                = "pe-sa-ip-export-blob-${local.short_workspace_id}"
+  resource_group_name = data.azurerm_resource_group.ws.name
+}
diff -Naur ../guacamole-azure-windowsvm/terraform/empty.txt terraform/empty.txt
--- ../guacamole-azure-windowsvm/terraform/empty.txt	1970-01-01 01:00:00.000000000 +0100
+++ terraform/empty.txt	2022-10-03 16:30:11.000000000 +0100
@@ -0,0 +1 @@
+The porter terraform mixin requires this directory to exist in the local build context.
diff -Naur ../guacamole-azure-windowsvm/terraform/variables.tf terraform/variables.tf
--- ../guacamole-azure-windowsvm/terraform/variables.tf	2022-10-03 16:28:13.000000000 +0100
+++ terraform/variables.tf	2022-10-03 16:30:11.000000000 +0100
@@ -4,5 +4,7 @@
 variable "tre_resource_id" {}
 variable "image" {}
 variable "vm_size" {}
-variable "shared_storage_access" {}
+variable "shared_storage_access" {
+  type = bool
+}
 variable "shared_storage_name" {}
diff -Naur ../guacamole-azure-windowsvm/terraform/windowsvm.tf terraform/windowsvm.tf
--- ../guacamole-azure-windowsvm/terraform/windowsvm.tf	2022-10-03 16:28:13.000000000 +0100
+++ terraform/windowsvm.tf	2022-10-03 16:30:11.000000000 +0100
@@ -11,6 +11,83 @@
   }
 }
 
+resource "azurerm_network_security_group" "vm_nsg" {
+  name                = "vm-nsg-${local.service_resource_name_suffix}"
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+  tags                = local.tre_user_resources_tags
+}
+
+resource "azurerm_network_security_rule" "allow_outbound_airlock_exip_storage_pe" {
+  access                        = "Allow"
+  # Should this be a list?
+  destination_address_prefixes  = [for pe in data.azurerm_private_endpoint_connection.airlock_export_inprogress_pe.private_service_connection: pe.private_ip_address]
+  destination_port_range        = "*"
+  direction                     = "Outbound"
+  name                          = "allow-airlock-exip-storage-pe"
+  network_security_group_name   = azurerm_network_security_group.vm_nsg.name
+  priority                      = 101
+  protocol                      = "*"
+  resource_group_name           = data.azurerm_resource_group.ws.name
+  source_address_prefixes       = azurerm_windows_virtual_machine.windowsvm.private_ip_addresses
+  source_port_range             = "*"
+}
+
+// Outbound traffic gets routed to the firewall
+resource "azurerm_network_security_rule" "allow_outbound_to_internet" {
+  access                      = "Allow"
+  destination_address_prefix  = "INTERNET"
+  destination_port_range      = "443"
+  direction                   = "Outbound"
+  name                        = "to-internet"
+  network_security_group_name = azurerm_network_security_group.vm_nsg.name
+  priority                    = 120
+  protocol                    = "Tcp"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_address_prefixes     = azurerm_windows_virtual_machine.windowsvm.private_ip_addresses
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "allow_outbound_webapps_to_vm" {
+  access = "Allow"
+  destination_port_ranges = [
+    "80",
+    "443",
+    "445",
+    "3306",
+    "3389",
+    "5432",
+  ]
+  destination_address_prefixes = azurerm_windows_virtual_machine.windowsvm.private_ip_addresses
+  source_address_prefixes      = data.azurerm_subnet.webapps.address_prefixes
+  direction                    = "Outbound"
+  name                         = "outbound-from-webapps-to-vm"
+  network_security_group_name  = azurerm_network_security_group.vm_nsg.name
+  priority                     = 140
+  protocol                     = "Tcp"
+  resource_group_name          = data.azurerm_resource_group.ws.name
+  source_port_range            = "*"
+}
+
+resource "azurerm_network_security_rule" "deny_outbound_override" {
+  access                      = "Deny"
+  destination_address_prefix  = "*"
+  destination_port_range      = "*"
+  direction                   = "Outbound"
+  name                        = "deny-outbound-override"
+  network_security_group_name = azurerm_network_security_group.vm_nsg.name
+  priority                    = 4096
+  protocol                    = "*"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_address_prefixes     = azurerm_windows_virtual_machine.windowsvm.private_ip_addresses
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_interface_security_group_association" "nsg_association" {
+  network_interface_id = azurerm_network_interface.internal.id
+  network_security_group_id = azurerm_network_security_group.vm_nsg.id
+}
+
 resource "random_string" "username" {
   length      = 4
   upper       = true
@@ -92,10 +169,10 @@
   template = file("${path.module}/vm_config.ps1")
   vars = {
     nexus_proxy_url     = local.nexus_proxy_url
-    SharedStorageAccess = tobool(var.shared_storage_access) ? 1 : 0
+    SharedStorageAccess = var.shared_storage_access ? 1 : 0
     StorageAccountName  = data.azurerm_storage_account.stg.name
     StorageAccountKey   = data.azurerm_storage_account.stg.primary_access_key
-    FileShareName       = data.azurerm_storage_share.shared_storage.name
+    FileShareName       = var.shared_storage_access ? data.azurerm_storage_share.shared_storage[0].name : ""
     CondaConfig         = local.image_ref[var.image].conda_config ? 1 : 0
   }
 }
@@ -106,6 +183,7 @@
 }
 
 data "azurerm_storage_share" "shared_storage" {
+  count                = var.shared_storage_access ? 1 : 0
   name                 = var.shared_storage_name
   storage_account_name = data.azurerm_storage_account.stg.name
 }
