diff -Naur ../base/terraform/airlock/data.tf terraform/airlock/data.tf
--- ../base/terraform/airlock/data.tf	2022-08-10 10:15:54.000000000 +0100
+++ terraform/airlock/data.tf	2022-09-14 23:46:54.000000000 +0100
@@ -23,3 +23,8 @@
   resource_group_name = local.core_resource_group_name
   namespace_name      = data.azurerm_servicebus_namespace.airlock_sb.name
 }
+
+data "azurerm_storage_account" "sa_import_inprogress" {
+  name                = local.import_in_progress_storage_name
+  resource_group_name = local.core_resource_group_name
+}
diff -Naur ../base/terraform/airlock/locals.tf terraform/airlock/locals.tf
--- ../base/terraform/airlock/locals.tf	2022-08-17 12:09:06.000000000 +0100
+++ terraform/airlock/locals.tf	2022-09-14 23:46:54.000000000 +0100
@@ -11,6 +12,8 @@
 
   # STorage AirLock IMport APProved
   import_approved_storage_name = lower(replace("stalimapp${substr(local.workspace_resource_name_suffix, -8, -1)}", "-", ""))
+  # STorage AirLock IMport InProgress
+  import_in_progress_storage_name = lower(replace("stalimip${var.tre_id}", "-", ""))
   # STorage AirLock EXport INTernal
   export_internal_storage_name = lower(replace("stalexint${substr(local.workspace_resource_name_suffix, -8, -1)}", "-", ""))
   # STorage AirLock EXport InProgress
diff -Naur ../base/terraform/airlock/storage_accounts.tf terraform/airlock/storage_accounts.tf
--- ../base/terraform/airlock/storage_accounts.tf	2022-08-17 12:09:06.000000000 +0100
+++ terraform/airlock/storage_accounts.tf	2022-09-14 23:46:54.000000000 +0100
@@ -48,6 +48,30 @@
   }
 }
 
+# 'In progress' storage account
+
+resource "azurerm_private_endpoint" "sa_import_inprogress_pe" {
+  name                = "stg-ip-import-blob-${local.workspace_resource_name_suffix}"
+  location            = var.location
+  resource_group_name = var.ws_resource_group_name
+  subnet_id           = var.services_subnet_id
+
+  lifecycle { ignore_changes = [tags] }
+
+  private_dns_zone_group {
+    name                 = "pdzg-stg-ip-import-blob-${local.workspace_resource_name_suffix}"
+    private_dns_zone_ids = [data.azurerm_private_dns_zone.blobcore.id]
+  }
+
+  private_service_connection {
+    name                           = "psc-stg-ip-import-blob-${local.workspace_resource_name_suffix}"
+    private_connection_resource_id = data.azurerm_storage_account.sa_import_inprogress.id
+    is_manual_connection           = false
+    subresource_names              = ["Blob"]
+  }
+
+  tags = var.tre_workspace_tags
+}
 
 # 'Drop' location for export
 resource "azurerm_storage_account" "sa_export_internal" {
diff -Naur ../base/terraform/storage.tf terraform/storage.tf
--- ../base/terraform/storage.tf	2022-08-10 10:15:54.000000000 +0100
+++ terraform/storage.tf	2022-09-14 23:46:54.000000000 +0100
@@ -9,17 +9,6 @@
   lifecycle { ignore_changes = [tags] }
 }
 
-resource "azurerm_storage_share" "shared_storage" {
-  name                 = "vm-shared-storage"
-  storage_account_name = azurerm_storage_account.stg.name
-  quota                = var.shared_storage_quota
-
-  depends_on = [
-    azurerm_private_endpoint.stgfilepe,
-    azurerm_storage_account_network_rules.stgrules
-  ]
-}
-
 resource "azurerm_storage_account_network_rules" "stgrules" {
   storage_account_id = azurerm_storage_account.stg.id
 
diff -Naur ../base/terraform/variables.tf terraform/variables.tf
--- ../base/terraform/variables.tf	2022-08-12 19:02:29.000000000 +0100
+++ terraform/variables.tf	2022-09-14 23:46:54.000000000 +0100
@@ -8,12 +8,6 @@
   description = "Resource ID"
 }
 
-variable "shared_storage_quota" {
-  type        = number
-  default     = 50
-  description = "Quota (in GB) to set for the VM Shared Storage."
-}
-
 variable "location" {
   type        = string
   description = "Azure location (region) for deployment of core TRE services"
